native/ffi/src/handler.rs

```diff
@@ -616,3 +794,216 @@ async fn is_post_shared_by_account(
     // TODO(shekohex): implement this function.
     Ok(ResponseBody::IsPostSharedByAccount(false))
 }
+
+async fn create_profile(
+    client: &Client<SubsocialRuntime>,
+    signer: &Signer,
+    CreateProfile { content }: CreateProfile,
+) -> Result<ResponseBody, Error> {
+    let content = match content {
+        Some(val) => val.into(),
+        None => {
+            return Err(Error {
+                kind: error::Kind::InvalidRequest.into(),
+                msg: String::from("Missing content value"),
+            })
+        }
+    };
+
+    let maybe_event = client
+        .create_profile_and_watch(signer, content)
+        .await?
+        .find_event::<ProfileCreatedEvent<_>>()?;
+    match maybe_event {
+        Some(event) => {
+            let body = ResponseBody::ProfileCreated(ProfileCreated {
+                account_id: event.account_id.to_ss58check_with_version(
+                    Ss58AddressFormat::SubsocialAccount,
+                ),
+            });
+            Ok(body)
+        }
+        None => Err(Error {
+            kind: error::Kind::NotFound.into(),
+            msg: String::from("Profile Created Event Not Found"),
```

Comment:
Maybe use event names in error messages as is? I mean full event names. Then it will be easier to search for them

```suggestion
            msg: String::from("ProfileCreated Event Not Found"),
```

Reply:
I will update that too while refactoring.